### PR TITLE
Add ratingByKeywordTest3 to the tester AND Add ratingCountPerProfTest3 to the tester

### PR DESCRIPTION
**Pull request by G1ngerAle**
check for abundant words in one sentence

### CODE REVIEW

**Comment by G1ngerAle** [comment]
round the average rating to two decimal number

File: FinalProject/StudentTests.java [hunk 1]
```diff
@@ -489,6 +489,32 @@ void ratingCountPerProfTest2() {
     assertEquals(v1, v2);
   }
 
+  // round the average rating to two decimal number
+  @Test
+  @DisplayName("RatingCountPerProf Test 3")
+  void ratingCountPerProfTest3() {
+    Parser parser = new Parser("");
```
Comment (SamZhang02) [inline FinalProject/StudentTests.java:496]: Can you add the fields here for the parser? Some people might query the fields hashmap
add at line 495:
```java
    HashMap<String, Integer> fields = new HashMap<>();
    fields.put("professor_name", 0);
    fields.put("school_name", 1);
    fields.put("department_name", 2);
    fields.put("post_date", 3);
    fields.put("student_star", 4);
    fields.put("student_difficult", 5);
    fields.put("comments", 6);
    fields.put("gender", 7);

    parser.fields = fields;
```

File: FinalProject/StudentTests.java [hunk 2]
```diff
@@ -574,6 +600,40 @@ void ratingByKeywordTest2() {
     assertEquals(v1, v2);
   }
 
+  // check for abundant words in one sentence
+  @Test
+  @DisplayName("RatingByKeyword Test 3")
```
Comment (SamZhang02) [inline FinalProject/StudentTests.java:605]: The data format here is wrong, can you change this test to this?
```java
  @Test
  @DisplayName("RatingByKeyword Test 3")
  void ratingByKeywordTest3() {
    Parser parser1 = new Parser("");
    Parser parser2 = new Parser("");
    ArrayList<String[]> dataArray1 = new ArrayList<>();
    ArrayList<String[]> dataArray2 = new ArrayList<>();

    HashMap<String, Integer> fields = new HashMap<>();
    fields.put("professor_name", 0);
    fields.put("school_name", 1);
    fields.put("department_name", 2);
    fields.put("post_date", 3);
    fields.put("student_star", 4);
    fields.put("student_difficult", 5);
    fields.put("comments", 6);
    fields.put("gender", 7);

    parser1.fields = fields;
    parser2.fields = fields;

    String[] arr1 = {"Diana  Oqimachi", "Long Beach City College", "Counseling department", "08/26/2014", "5", "1", "fun"};
    String[] arr2 = {"Diana  Oqimachi", "Long Beach City College", "Counseling department", "08/26/2014", "5", "1", "Fun fun fun!"};
    dataArray1.add(arr1);
    dataArray2.add(arr2);

    parser1.data = dataArray1;
    parser2.data = dataArray2;

    DataAnalyzer analyzer1 = new RatingByKeyword(parser1);
    DataAnalyzer analyzer2 = new RatingByKeyword(parser2);

    MyHashTable<String, Integer> output1 = analyzer1.getDistByKeyword("fun");
    MyHashTable<String, Integer> output2 = analyzer2.getDistByKeyword("fun");

    ArrayList<String> k1 = output1.getKeySet();
    ArrayList<String> k2 = output2.getKeySet();
    ArrayList<Integer> v1 = output1.getValueSet();
    ArrayList<Integer> v2 = output2.getValueSet();

    Collections.sort(k1);
    Collections.sort(k2);
    Collections.sort(v1);
    Collections.sort(v1);

    assertEquals(k1, k2);
    assertEquals(v1, v2);
  }
```
I just corrected the data format and added fields as well

**Comment by SamZhang02** [comment]
~~It would be great if you can just resolve to reviews above. Otherwise LGTM.~~

Btw, i'll hop on a flight to China in a few hours, so i might reply to your PRs really slowly from tonight to a day or two later. If you happen to add more tests, you can tag @ssantoshp to review/merge

You can also add me on WeChat or discord if you wanna talk tests: 
- WeChat: saamenerve
- Discord: saamenerve#2510

**Review by SamZhang02 — COMMENTED** [review]
All good for the review, i changed them myself and merged, sorry for the trouble

**Comment by G1ngerAle** [comment]
No worries. Bon voyage!

**Comment by santoshlite** [comment]
Sure @SamZhang02, have a safe trip!

**Comment by SamZhang02** [comment]
Thanks guys!

**Pull request merged**